workers/varcode_annotator.py

```diff
@@ -0,0 +1,133 @@
+            write_to_table_via_csv(tmp_table, rows=gene_names, connection=con)
+            # Add gene names from temp table to genotypes.
+            (genotypes.update()
+             .where(genotypes.c.contig == tmp_table.c.contig)
+             .where(genotypes.c.position == tmp_table.c.position)
+             .where(genotypes.c.reference == tmp_table.c.reference)
+             .where(genotypes.c.alternates == tmp_table.c.alternates)
+             .where(genotypes.c.vcf_id == vcf_id)
+             .values(
+                 {
+                    'annotations:varcode_gene_name': tmp_table.c.gene_name,
+                    'annotations:varcode_transcript': tmp_table.c.transcript,
+                    'annotations:varcode_effect_notation': tmp_table.c.notation,
+                    'annotations:varcode_effect_type': tmp_table.c.effect_type
+                 }
+             )).execute()
+        finally:
+            con.execute("DISCARD TEMP")
+
+        # We've added annotations:varcode_*, so update the columns to display.
+        update_extant_columns(metadata, con, vcf_id)
+        return(vcf_id)
+
+
+def write_to_table_via_csv(table, rows, connection):
+    with temp_csv(mode='r+', tmp_dir=TEMPORARY_DIR) as csv_file:
+        # Don't use commas as a delim, as commas are part of gene_names.
+        csv.writer(csv_file, delimiter='\t').writerows(rows)
+        csv_file.seek(0, 0)
+        connection.connection.cursor().copy_from(csv_file, sep='\t', null='',
+                                                 table=table.name)
+        connection.connection.commit()
+
+
+def get_varcode_annotations(genotypes, vcf_id, ensembl_release_num):
+    """Get contig, position, ref and alt data from the genotypes table, 
+    and get the best effect from Varcode library. Return a list of the form:
+    [[contig, position, "NAME,NAME,..."], [contig...], ...]
+    """
+    results = select([
+            genotypes.c.contig,
+            genotypes.c.position,
+            genotypes.c.reference,
+            genotypes.c.alternates
+        ]).where(genotypes.c.vcf_id == vcf_id).execute()
+
+    ensembl_rel = EnsemblRelease(ensembl_release_num)
+
+    varcode_annotations = []
+    for contig, position, reference, alternates in results:
+        variant = Variant(
+            contig=contig, 
+            start=position, 
+            ref=reference.encode('ascii','ignore'),
+            alt=alternates.encode('ascii','ignore'), 
+            ensembl=ensembl_rel)
+
+        # This will give us a single, yet relevant effect
+        best_effect = variant.effects().top_priority_effect()
+        gene_name = best_effect.gene_name
+        transcript = best_effect.transcript_id
+        notation = best_effect.short_description
+        effect_type = type(best_effect).__name__
+        # Make it human readable
+        effect_type = re.sub("([a-z])([A-Z])","\g<1> \g<2>", effect_type)
+        varcode_annotations.append([contig, position, reference, alternates, 
```

Comment:
Indentation should be pep8-y cf. http://legacy.python.org/dev/peps/pep-0008/#indentation

(Need to line up the hanging indent, in this case)